src/debugging/DockerDebugConfigurationProvider.ts

```diff
@@ -111,27 +113,60 @@ export class DockerDebugConfigurationProvider implements DebugConfigurationProvi
             // the launch configuration, and since this method comes later then we shouldn't remove a container
             // that we just created.
             // TODO: this needs to be removed as soon as the Python extension adds a way to retry while connecting to a remote debugger.
             if (resolvedConfiguration.type !== 'python') {
                 try {
-                    await this.dockerClient.removeContainer(resolvedConfiguration.dockerOptions.containerNameToKill, { force: true });
+                    await this.dockerClient.removeContainer(resolvedConfiguration.dockerOptions.containerName, { force: true });
                 } catch { }
             }
 
             // Now register the container for removal after the debug session ends
             const disposable = debug.onDidTerminateDebugSession(async session => {
                 const sessionConfiguration = <ResolvedDebugConfiguration>session.configuration;
 
-                if (sessionConfiguration
-                    && sessionConfiguration.dockerOptions
-                    && sessionConfiguration.dockerOptions.containerNameToKill === resolvedConfiguration.dockerOptions.containerNameToKill) {
+                if (sessionConfiguration?.dockerOptions?.containerName === resolvedConfiguration.dockerOptions.containerName) {
                     try {
-                        await this.dockerClient.removeContainer(resolvedConfiguration.dockerOptions.containerNameToKill, { force: true });
+                        await this.dockerClient.removeContainer(resolvedConfiguration.dockerOptions.containerName, { force: true });
+                    } finally {
+                        disposable.dispose();
+                    }
+                } else {
+                    return; // Return without disposing--this isn't our debug session
+                }
+            });
+        }
+    }
+
+    private async registerOutputPortsAtDebugging(resolvedConfiguration: ResolvedDebugConfiguration): Promise<void> {
+        if (resolvedConfiguration?.dockerOptions?.containerName) {
+            const disposable = debug.onDidStartDebugSession(async session => {
+                const sessionConfiguration = <ResolvedDebugConfiguration>session.configuration;
+
+                if (sessionConfiguration?.dockerOptions?.containerName === resolvedConfiguration.dockerOptions.containerName) {
+                    try {
+                        const inspectInfo = await ext.dockerode.getContainer(resolvedConfiguration.dockerOptions.containerName)?.inspect();
+                        const portMappings: string[] = [];
+
+                        if (inspectInfo?.NetworkSettings?.Ports) {
+                            for (const containerPort of Object.keys(inspectInfo.NetworkSettings.Ports)) {
+                                const mapping = inspectInfo.NetworkSettings.Ports[containerPort][0];

```

Comment:
Do we care about more than just the first mapping (which would be unusual)?

Reply:
Good thought, I made that change.